world/mujoco_ros/src/mushr_visualize.cpp

```diff
@@ -34,24 +35,27 @@ int main(int argc, char** argv)
     ROS_ERROR("Failed to get param 'save_trajectory'.");
   }
 
-  const std::string root{ ros::this_node::getNamespace() };
   mj_ros::SimulatorPtr sim{ mj_ros::simulator_t::initialize(model_path, save_trajectory) };
   controller_listener_t<CtrlMsg, PlanMsg> controller_listener(n, sim->d);
+  mj_ros::ground_truth_pose_t<prx_models::MushrObservation> ground_truth_pose(n, sim, 15);
 
   ros::Subscriber reset_subscriber;
   reset_subscriber = n.subscribe(root + "/reset", 1000, &mj_ros::simulator_t::reset_simulation, sim.get());
 
   // Set the threads
   std::thread step_thread(&mj_ros::simulator_t::run, &(*sim));  // Mj sim
+  std::thread publisher_thread(&mj_ros::ground_truth_pose_t<prx_models::MushrObservation>::run, &ground_truth_pose);
   mj_ros::simulator_visualizer_t visualizer{ sim };             // Mj Viz
   ros::AsyncSpinner spinner(1);                                 // 1 thread for the controller
```

Comment:
You can use [this](https://github.com/PRX-Kinodynamic/mujoco_ros/blob/921d525d982aca2f7f13b986c673e44eb2dfe876/world/mujoco_ros/include/mujoco_ros/simulator.hpp#L293C8-L293C8) function like [here](https://github.com/PRX-Kinodynamic/mujoco_ros/blob/921d525d982aca2f7f13b986c673e44eb2dfe876/world/mujoco_ros/src/mushr_feedback.cpp#L41). This already takes care of _spinning_ for topics/services